src/extensions/unit/unitext_hooks.cpp

```diff
@@ -680,6 +681,285 @@ DECLARE_PATCH(_UnitClass_Per_Cell_Process_AutoHarvest_Assign_Harvest_Mission_Pat
+ *
+ *  @author: Rampastring
+ */
+DECLARE_PATCH(_UnitClass_Try_To_Deploy_Transform_To_Vehicle_Patch) {
+    GET_REGISTER_STATIC(UnitClass*, this_ptr, esi);
+    static int address;
+    address = _UnitClass_Try_To_Deploy_Transform_To_Vehicle_Patch_Func(this_ptr);
+    JMP(address);
+}
+
+
+/**
+ *  #issue-715
+ *
+ *  Hack to display the the correct cursor for transformable units
+ *  upon ACTION_SELF.
+ *
+ *  @author: Rampastring
+ */
+DECLARE_PATCH(_UnitClass_What_Action_Self_Check_For_Vehicle_Transform_Patch) {
+    GET_REGISTER_STATIC(UnitClass*, this_ptr, esi);
+    static UnitTypeClass* unittype;
+    static UnitTypeClassExtension* unittypeext;
+
+    unittype = this_ptr->Class;
+    unittypeext = Extension::Fetch<UnitTypeClassExtension>(unittype);
+
+    /**
+     *  Stolen bytes/code.
+     */
+    if (unittype->DeploysInto != nullptr) {
+        goto original_code;
+    }
+
+    /**
+     *  Check if this unit is able to transform into another unit.
+     *  If not, we don't have anything else to do here.
+     */
+    if (unittypeext->TransformsInto == nullptr) {
+        goto no_deploy_action;
+    }
+
+    /**
+     *  If this unit is able to transform to a different unit, check if it requires charge for it.
+     *  If it does, then check whether we have enough charge.
+     */
+    if (unittypeext->IsTransformRequiresFullCharge && this_ptr->CurrentCharge < this_ptr->Class->MaxCharge) {
+
+        /**
+         *  We don't have enough charge!
+         */
+        goto return_no_deploy;
+
+    } else if (this_ptr->entry_2A4()) {
+
+        /**
+         *  The unit is dying or under an EMP effect, don't allow it to transform.
+         */
+        goto return_no_deploy;
+    }
+
+
+    /**
+     *  We can transform and either don't need charge or we have enough of it, return ACTION_SELF
+     */
+return_self:
+    _asm { pop edi }
+    _asm { pop esi }
+    _asm { pop ebp }
+    _asm { pop ebx }
+    _asm { add esp, 10h }
+    _asm { mov eax, ACTION_SELF }
+    _asm { retn 8 }
+
+    /**
+     *  We don't have enough charge, return ACTION_NO_DEPLOY
+     */
+return_no_deploy:
+    _asm { pop edi }
+    _asm { pop esi }
+    _asm { pop ebp }
+    _asm { pop ebx }
+    _asm { add esp, 10h }
+    _asm { mov eax, ACTION_NO_DEPLOY }
+    _asm { retn 8 }
+
+
+    /**
+     *  Check whether the unit can deploy.
+     */
+original_code:
+    JMP(0x0065602B);
+
+    /**
+     *  The unit can neither deploy nor transform.
+     *  Continue function execution after the deployable check.
+     */
+no_deploy_action:
+    _asm{ mov eax, unittype }
+    JMP_REG(ecx, 0x00656344);
+}
+
+
+/**
+ *  #issue-715
+ *
+ *  Check whether the unit is able to transform into another unit
+ *  when performing the "Unload" mission.
+ *
+ *  @author: Rampastring
+ */
+DECLARE_PATCH(_UnitClass_Mission_Unload_Transform_To_Vehicle_Patch) {
+    GET_REGISTER_STATIC(UnitTypeClass*, unittype, eax);
```

Comment:
Needs a space between `UnitTypeClass` and `*`.

This is most likely your MSVC auto-formatting taking over here, you can change those easily in Tools -> Options -> Text Editor -> C/C++.